easybuild/easyblocks/i/impi.py

```diff
@@ -139,15 +140,17 @@ def post_install_step(self):
 
         impiver = LooseVersion(self.version)
         if impiver == LooseVersion('4.1.1.036') or impiver >= LooseVersion('5.0.1.035'):
+            if impiver >= LooseVersion('2018.0.128'):
+                script_paths = [os.path.join('intel64', 'bin')]
+            else:
+                script_paths = [os.path.join('intel64', 'bin'), os.path.join('mic', 'bin')]
             # fix broken env scripts after the move
-            for script in [os.path.join('intel64', 'bin', 'mpivars.csh'), os.path.join('mic', 'bin', 'mpivars.csh')]:
-                for line in fileinput.input(os.path.join(self.installdir, script), inplace=1, backup='.orig.easybuild'):
-                    line = re.sub(r"^setenv I_MPI_ROOT.*", "setenv I_MPI_ROOT %s" % self.installdir, line)
-                    sys.stdout.write(line)
-            for script in [os.path.join('intel64', 'bin', 'mpivars.sh'), os.path.join('mic', 'bin', 'mpivars.sh')]:
-                for line in fileinput.input(os.path.join(self.installdir, script), inplace=1, backup='.orig.easybuild'):
-                    line = re.sub(r"^I_MPI_ROOT=.*", "I_MPI_ROOT=%s; export I_MPI_ROOT" % self.installdir, line)
-                    sys.stdout.write(line)
+            regex_subs = (r"^setenv I_MPI_ROOT.*", r"setenv I_MPI_ROOT %s" % self.installdir)
```

Comment:
@damianam This has to be a list:

```python
regex_subs = [(..., ...)]
```

same below

Otherwise:

```
  File "/easybuild/easyblocks/i/impi.py", line 150, in post_install_step
    apply_regex_substitutions(os.path.join(self.installdir, script), regex_subs)
  File "/easybuild/tools/filetools.py", line 906, in apply_regex_substitutions
    for i, (regex, subtxt) in enumerate(regex_subs):
ValueError: too many values to unpack
```